File: GFG/Stacks/arrayimp.py
class Stack:
    def __init__(self):
        self.arr=[]
        self.sz=0
    def push(self,data):
        self.arr.append(data)
        self.sz+=1
    def pop(self):
        res= self.arr[self.sz-1]
        self.arr.pop(-1)
        self.sz-=1
        return res
    def top(self):
        return self.arr[-1]
    def isEmpty(self):
        return self.sz==0
    def size(self):
        return self.sz
    
q=Stack()
for i in range(1,6):
    q.push(i*12)
print(q.size())
while(q.isEmpty()==False):
    print(q.pop())
    

File: GFG/Stacks/arrayimp.cpp
#include<iostream>
#include<vector>
using namespace std;
template <typename T>
class Stack{
    public:
    vector<T> arr;
    int sz=0;
    void push(T value){
        arr.push_back(value);
        sz++;
    }
    T pop(){
        T res=arr[sz-1];
        sz--;
        arr.pop_back();
        return res;
    }
    T top(){
        return arr.back();
    }
    bool isEmpty(){
        return sz==0;
    }
    int size(){
        return sz;
    }
};

int main(){
    Stack<string> q;
    q.push("qaz");
    q.push("wsx");
    q.push("edc");
    q.push("rfv");
    cout<<q.size()<<endl;
    cout<<q.top()<<endl;
    cout<<q.isEmpty()<<endl;
    while(q.isEmpty()!=1){
        cout<<q.pop()<<"||"<<endl;
    }
    return 0;
}